specs/eid-hermes-commands-format.md

```diff
@@ -0,0 +1,214 @@
+Some parts of Command Entries and Responses are command-specific and are
+described below.
+
+| Bytes | Description                              |
+|-------|------------------------------------------|
+| 00    | Opcode: defines which command to execute |
+| 01    | Reserved                                 |
+| 03:02 | Command Identifier: a Hermes user should never repeat this field. Different users may repeat an ID (the driver has other ways to differentiate between users). It is undefined behaviour to reuse an ID. |
+| 07:04 | Reserved                                 |
+| 31:08 | Command Specific                         |
+
+**Table 1: Submission Entry format**
+
+| Bytes | Description        |
+|-------|--------------------|
+| 01:00 | Command Identifier |
+| 02    | Status             |
+| 07:03 | Reserved           |
+| 15:08 | Command Specific   |
+
+**Table 2: Completion Entry format**
+
+The following `Status` of CEs are defined:
+
+| Status | Description          |
+|--------|----------------------|
+| 0x00   | Success              |
+| 0x01   | Not enough space     |
+| 0x02   | Invalid Program Slot |
+| 0x03   | Invalid Data Slot    |
+| 0x04   | Invalid Address      |
+| 0x05   | eBPF error           |
+
+**Table 3: Status**
+
+## List of Commands
+
+The following commands are defined for Hermes devices:
+
+| Opcode | Command             |
+|--------|---------------------|
+| 0x00   | Request Slot        |
+| 0x01   | Release Slot        |
+| 0x10   | Write Data to Slot  |
+| 0x11   | Read Data from Slot |
+| 0x80   | Run Program         |
+
+**Table 4: List of commands**
+
+### Request Slots (Opcode 0x00)
+
+This command can be used to reserve a program or data slot on the device.
+
+| Bytes | Description     |
+|-------|-----------------|
+| 08    | Slot type       |
+
+**Table 5: Request Slots Submission Entry**
+
+| Bytes | Description                                          |
+|-------|------------------------------------------------------|
+| 08    | ID of allocated slot. Only valid when Status is 0x00 |
+
+**Table 6: Request Slots Completion Entry**
+
+This command may return the following status:
+
+| Status | Description                            |
+|--------|----------------------------------------|
+| 0x00   | Success                                |
+| 0x01   | All slots of requested type are in use |
+
+**Table 7: Request Slots status codes**
+
+### Release Slots (Opcode 0x01)
+
+This command can be used to release a program or data slot on the device.
+
+It is invalid to release a slot that is not allocated.
+
+| Bytes | Description |
+|-------|-------------|
+| 08    | Slot type   |
+| 09    | Slot ID     |
+
+**Table 8: Release Slots Submission Entry**
+
+| Bytes | Description                |
+|-------|----------------------------|
+
+(no command-specific bytes used)
+
+**Table 9: Release Slots Completion Entry**
+
+This command may return the following status:
+
+| Status | Description                             |
```

Comment:
Same values used more than once?

Reply:
Yes. If you look at Table 3, I have defined 0x02 as "Invalid Program Slot", which could be for the two possible reasons defined in this table. I could make it in a single line if it makes it clearer.